src/main/java/com/example/lures/LureService.java

```diff
@@ -35,4 +35,22 @@ public Lure insert(String product, String company, double size, double weight) {
         lureMapper.insert(lure);
         return lure;
     }
+
+    public Lure update(Integer id, String product, String company, double size, double weight) {
+        Optional<Lure> optionalLure = this.lureMapper.findById(id);
+        Lure lure = optionalLure.orElseThrow(() -> new LureNotFoundException("Lure not found"));
+
+        List<Lure> existingLureId = lureMapper.findByLure(product);

```

Comment:
lureMapper.findByLure(product)の検索結果ですが product は UNIQUE なので１件のみしか HIT しないはず。
であれば List でなく Optional を使うのが適当かと思います。
```suggestion
        Optional<Lure> existingLureId = lureMapper.findByLure(product);

```